web/src/app/services/task/task.service.ts

```diff
@@ -69,6 +69,25 @@ export class TaskService {
     return new Task(taskId, type, label, required, index, multipleChoice);
   }
 
+  /**
+   * Returns a new task which is an exact copy of the provided task, but with new UUIDs for all items recursively.
+   */
+  createDuplicateTask(task: Task): Task {
+    return {
+      ...task,
+      id: this.dataStoreService.generateId(),
+      multipleChoice: task.multipleChoice
+        ? ({
+            ...task.multipleChoice,
+            options: task.multipleChoice.options?.map(option => ({
```

Comment:
Nit: For consistency and readability, you could define this in its own method, `duplicateMultipleChoiceOption`?